File: IOS/Sources/OpenRoaming/OpenRoaming/Models/UserDetail.swift
//
//  UserDetail.swift
//  OpenRoaming
//
//  Created by olivier duque campos on 17/06/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

import Foundation
import UIKit

public struct UserDetail {
    
    public var name: String
    public var phone: String
    public var email: String
    public var age: Int64
    public var zipCode: String

    public init(data: Data) throws {
        let jsonResult: NSDictionary = try (JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? NSDictionary)!
        self.name = jsonResult["name"] as? String ?? "Unknown name"
        self.phone = jsonResult["phone"] as? String ?? ""
        self.email = jsonResult["email"] as? String ?? ""
        self.age = jsonResult["age"] as? Int64 ?? 0
        self.zipCode = jsonResult["zip"] as? String ?? ""
    }
    
    public func json() -> Data {
        let jsonValues: NSMutableDictionary = NSMutableDictionary()
        jsonValues.setValue(self.name, forKey: "name")
        jsonValues.setValue(self.email, forKey: "email")
        jsonValues.setValue(self.phone, forKey: "phone")
        jsonValues.setValue(self.age, forKey: "age")
        jsonValues.setValue(self.zipCode, forKey: "zip")
        
        return try! JSONSerialization.data(withJSONObject: jsonValues)
    }
}

File: Android/src/main/java/com/cisco/or/sdk/models/UserDetail.kt
package com.cisco.or.sdk.models

import com.cisco.or.sdk.exceptions.EmailException
import com.cisco.or.sdk.exceptions.PhoneNumberException
import com.cisco.or.sdk.utils.ValidateUtil
import org.json.JSONObject
import java.lang.Exception
import java.util.*

data class UserDetail(val name: String, val phone: String, val email: String, val age: Int, val zipCode: String ) {

    init {
        if(phone.isNotEmpty() && !ValidateUtil.validatePhoneNumber(phone)){
            throw PhoneNumberException()
        }
        if(email.isNotEmpty() && !ValidateUtil.validateEmail(email)){
            throw EmailException()
        }
    }

}

fun UserDetail(jsonObject: JSONObject): UserDetail{
    val name = try { jsonObject.getString("name") }catch (e:Exception) {""}
    val phone = try { jsonObject.getString("phone") }catch (e:Exception) {""}
    val email = try { jsonObject.getString("email") }catch (e:Exception) {""}
    val age = try { jsonObject.getInt("age") }catch (e:Exception) {0}
    val zip = try { jsonObject.getString("zip") }catch (e:Exception) {""}
    return UserDetail(name, phone, email, age, zip)
}